project-koku/koku
# [COST-2225] Source stats only show information for current and previous month

Fixes [COST-2225](https://issues.redhat.com/browse/COST-2225)

Changes proposed in this PR:
/api/cost-management/v1/sources/{source_uuid}/stats/  Used to report only the two previous months, now it reports all previous months in the data


Testing Instructions:
1. Make sure you have these params in your local env
INITIAL_INGEST_NUM_MONTHS >= 3 and RETAIN_NUM_MONTHS >= 3
2.  Edit the data generation in `iqe_cost_management/data/aws/aws_static_report_basic.yml` and replace every `start_date: last_month` with `start_date: 2022-04-01 `(or the start date from 3 months ago.
3. Edit the test to add a break point, for this test we will edit `test_api_aws_ingest_single_source`
4. run the test with the command in your iqe env `ENV_FOR_DYNACONF=local iqe tests plugin cost_management -k "test_api_aws_ingest_single_source" -vv --pdb`
5. Hit http://localhost:8000/api/cost-management/v1/reports/aws/costs/?start_date=2022-04-01&end_date=2022-04-30
6. Take a source uuid from that end point and plug in into http://localhost:8000/api/cost-management/v1/sources/{source_uuid}/stats/ 
7. Verify that the april data is there
---
Additional Context: Please reach out to me for more information how to tet.
